src/test/java/com/cloudant/tests/CloudantClientTests.java

```diff
@@ -101,4 +103,19 @@ public void cookieNegativeTest() {
         }
     }
 
+    /**
+     * Assert that the User-Agent header is of the expected form.
+     * It would be nice to test that the header is actually added to requests, but we can't
+     * assert that without making assumptions about the underlying client or asserting server-side.

```

Comment:
Note @brynh's use of a mock HTTP server in the certificate tests if you want to do the full thing here.